bigframes/_config/bigquery_options.py

```diff
@@ -252,6 +261,8 @@ def allow_large_results(self) -> bool:
 
     @allow_large_results.setter
     def allow_large_results(self, value: bool):
+        if not value:
+            warnings.warn(LARGE_RESULTS_WARNING_MESSAGE, UserWarning)
```

Comment:
I don't think the setter makes the most sense for this warning. We want to encourage people to use this, not discourage them. I'd rather see it when they call a function that initiates `query_and_wait` and sampling has been requested.